File: oop-task/6-polymorphism.cpp
#include<bits/stdc++.h>
using namespace std;

// Base class
class Animal {
  public:
    void animalSound() 
    {
      cout << "The animal will make a sound. \n";
    }
};

// Derived class
class Cat : public Animal {
  public:
    void animalSound() {
      cout << "The Cat will say Mew Mew \n";
    }
};

// Derived class
class Dog : public Animal {
  public:
    void animalSound() {
      cout << "The dog will say bow wow \n";
    }
};

int main() {
  Animal myAnimal;
  Cat myCat;
  Dog myDog;

  myAnimal.animalSound();
  myCat.animalSound();
  myDog.animalSound();
  
  return 0;
}

File: oop-task/6-polymorphism.py
class Animal:
    def animalSound(self):
        return "Animal sound default"

class Dog(Animal):
    def animalSound(self):
        return "Woof!"

class Cat(Animal):
    def animalSound(self):
        return "Meow!"

class Snake(Animal):
    def animalSound(self):
        return "Fssss!"

def make_animal_speak(animal):
    return animal.animalSound()

dog = Dog()
cat = Cat()
snake = Snake()

print(make_animal_speak(dog))  
print(make_animal_speak(cat))   
print(make_animal_speak(duck))